packages/xdl/src/Project.ts

```diff
@@ -76,16 +78,85 @@ const treekillAsync = promisify(treekill);
 const ngrokConnectAsync = promisify(ngrok.connect);
 const ngrokKillAsync = promisify(ngrok.kill);
 
-type CachedSignedManifest = {
-  manifestString: ?string,
-  signedManifest: ?string,
-};
+type CachedSignedManifest =
+  | {
+      manifestString: null;
+      signedManifest: null;
+    }
+  | {
+      manifestString: string;
+      signedManifest: string;
+    };
 
 let _cachedSignedManifest: CachedSignedManifest = {
   manifestString: null,
   signedManifest: null,
 };
 
+type Asset =
```

Comment:
One thing that might be confusing is that there are multiple types named `Asset` in the code. Perhaps more descriptive names like AssetInfo would help disambiguate them.

Reply:
[We already have `AssetInfo`](https://github.com/expo/expo-cli/pull/945/files#diff-8d46d7b06777595a524a77ee67736074R45) 😄 

I've renamed the `Asset` in schemer to  a more descriptive name `AssetField`.